src/webots/nodes/utils/WbNodeUtilities.cpp

```diff
@@ -929,12 +929,19 @@ bool WbNodeUtilities::isInBoundingObject(const WbNode *node) {
   const WbNode *const p = node->parentNode();
   if (p) {
     const WbSolid *const s = dynamic_cast<const WbSolid *>(p);
+    const WbFluid *const f = dynamic_cast<const WbFluid *>(p);
+    const WbNode *boundingObject = NULL;
     if (s)
-      return s->boundingObject() == node;
+      boundingObject = s->boundingObject();
+    else if (f)
+      boundingObject = f->boundingObject();
 
-    const WbFluid *const f = dynamic_cast<const WbFluid *>(p);
-    if (f)
-      return f->boundingObject() == node;
+    while (boundingObject) {
+      if (boundingObject == node)
+        return true;
+      else
+        boundingObject = boundingObject->protoParameterNode();
```

Comment:
`else` is useless given the previous return
```suggestion
      boundingObject = boundingObject->protoParameterNode();
```